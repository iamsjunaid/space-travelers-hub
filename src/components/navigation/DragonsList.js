import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchDragons } from '../../redux/dragons/dragonsSlice';
import Dragons from '../Dragons';

function DragonsList() {
  const { dragons, isLoading } = useSelector((state) => state.dragons);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchDragons());
  }, [dispatch]);

  if (isLoading) {
    return <p className="loading">Loading...</p>;
  }

  if (!dragons || dragons.length === 0) {
    return <p>No dragons found.</p>;
  }

  return (
    <div>
      <ul>
        {dragons.map((dragon) => (
          <Dragons
            key={dragon.id}
            id={dragon.id}
            name={dragon.name}
            type={dragon.type}
            image={dragon.image}
            description={dragon.description}
            reserved={dragon.reserved}
          />
        ))}
      </ul>
    </div>
  );
}
export default DragonsList;
