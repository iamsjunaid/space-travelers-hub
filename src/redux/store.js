import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';
import dragonsReducer from './dragons/dragonsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
    dragons: dragonsReducer,
  },
});

export default store;
