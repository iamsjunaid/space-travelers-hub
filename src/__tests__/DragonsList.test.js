import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../redux/store';
import DragonsList from '../components/navigation/DragonsList';

test('renders DragonsList component correctly', () => {
  const dragons = [
    {
      id: '1',
      name: 'Dragon Name 1',
      type: 'Dragon Type 1',
      image: 'dragon1.jpg',
      description: 'Dragon Description 1',
      reserved: false,
    },
    {
      id: '2',
      name: 'Dragon Name 2',
      type: 'Dragon Type 2',
      image: 'dragon2.jpg',
      description: 'Dragon Description 2',
      reserved: true,
    },
  ];

  const { container } = render(
    <Provider store={store}>
      <DragonsList dragons={dragons} />
    </Provider>,
  );
  expect(container.innerHTML).toMatchSnapshot();
});
