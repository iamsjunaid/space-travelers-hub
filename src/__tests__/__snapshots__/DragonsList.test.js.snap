// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DragonsList component correctly 1`] = `"<p class=\\"loading\\">Loading...</p>"`;
